lua/luau-lsp/studio.lua

```diff
@@ -0,0 +1,144 @@
+-- Acknowledgements:
+-- https://github.com/zilibobi/luau-tree.nvim/tree/main
+-- https://github.com/JohnnyMorganz/luau-lsp/blob/main/editors/code/src/extension.ts
+-- https://github.com/neovim/neovim/blob/master/runtime/lua/vim/lsp/rpc.lua
+
+local compat = require "luau-lsp.compat"
+local config = require "luau-lsp.config"
+local http = require "luau-lsp.http"
+local log = require "luau-lsp.log"
+local uv = compat.uv
+
+local is_listening = false
+
+local server
+local socket
+
+local data_model
+local current_port = config.get().plugin.port
+
+local function send_status(target_socket, status, body)
+  local response = http.create_response({}, body or "", status)
+  target_socket:write(response)
+end
+
+local M = {}
+
+local function start_server(port)
+  assert(type(port) == "number", "Port must be a number")
+  assert(is_listening == false, "This server object is already bound to http://localhost:" .. port)
+
+  is_listening = true
+  current_port = port
+
+  server = uv.new_tcp()
+
+  server:bind("127.0.0.1", port)
+
+  server:listen(128, function(listen_err)
+    assert(not listen_err, listen_err)
+
+    log.info("Now listening on port " .. port)
```

Comment:
This should be right after `server:bind(...)`. but we can have another message here, maybe something like "studio connected on port ..."

Reply:
I added "Studio connect on port" under `socket:read_start(...)` instead so it logs whenever the plugin connects